doc/modulesync.rst

```diff
@@ -0,0 +1,88 @@
+# Copyright (C) 2021  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+
+====================
+DNF modulesync Plugin
+====================
+
+Download packages from modules and/or create a repository with modular data.
+
+--------
+Synopsis
+--------
+
+``dnf modulesync [options] <module-spec>...``
+
+-----------
+Description
+-----------
+
+`modulesync` downloads packages from modules according to provided arguments and creates a repository with modular data
+in working directory. Im modular environment it is recommend to use the command in combination with other dnf commands
+which downloads RPM packages. See examples
```

Comment:
I'm a bit confused here. Could you elaborate in more details the use case when user needs to download rpm packages separately (using `dnf download` or such) and then run `modulesync` to generate modular metadata? I thought that `modulesync` will download rpm packages from given modules.

Reply:
In case that a package has modular label in rpm header it cannot be used in new repository without modular metadata. module sync will create a repository including modular metadata.

Reply:
But modulesync also downloads the rpm package itself, doesn't it? What I do not understand is the need of `dnf download` before `dnf modulesync`. I really think this deserves more detailed documentation in the man pages.

Reply:
Good point